pandas/tseries/tests/test_timeseries.py

```diff
@@ -5393,6 +5394,73 @@ def test_day_not_in_month_ignore(self):
             '2015-04-31', errors='ignore', format="%Y-%m-%d"), '2015-04-31')
 
 
+class TestAstype(tm.TestCase):
+
+    def test_datetimeindex(self):
+        # GH 13149, GH 13209
+        idx = DatetimeIndex(['2016-05-16', 'NaT', NaT, np.NaN])
+
+        result = idx.astype(object)
+        expected = Index([Timestamp('2016-05-16')] + [NaT] * 3, dtype=object)
+        tm.assert_index_equal(result, expected)
+
+        result = idx.astype(int)
+        expected = Int64Index([1463356800000000000] +
+                              [-9223372036854775808] * 3, dtype=np.int64)
+        tm.assert_index_equal(result, expected)
+
+        result = idx.astype(str)
```

Comment:
Here, Python2 returns an Index with mixed type: Date '2016-05-16' is a string, and NaT's are unicode. Is it all right?